crates/qasm3/Cargo.toml

```diff
@@ -0,0 +1,20 @@
+[package]
+name = "qiskit-qasm3"
+version.workspace = true
+edition.workspace = true
+rust-version.workspace = true
+license.workspace = true
+
+[lib]
+name = "qiskit_qasm3"
+crate-type = ["cdylib"]
+
+[features]
+# This is a test-only shim removable feature.  See the root `Cargo.toml`.
+default = ["extension-module"]
+extension-module = ["pyo3/extension-module"]
+
+[dependencies]
+pyo3.workspace = true
+hashbrown.version = "0.14.0"
+oq3_semantics = { git = "https://github.com/Qiskit/openqasm3_parser.git", rev = "57407ee" }
```

Comment:
The base parser isn't released yet, and _technically_, we don't actually need it to be if we specify the dependency as a git revision.  It's kind of like using a submodule, but easier for users.

That said, we totally should release it and fix this dependency before we release Qiskit properly, but for now, this works and is convenient to update.